data_structures/trie_using_hashmap.cpp

```diff
@@ -0,0 +1,337 @@
+                return;
+            }
+            if (curr->word_end) {
+                cnt++;
+            }
+
+            nodes.push(curr->children[ch]);
+            curr = curr->children[ch];
+        }
+        /*delete only when its a word and it has no children after and also no
+         * prefix in the line*/
+        if (nodes.top()->word_end) {
+            nodes.top()->word_end = false;
+        }
+        while (!(nodes.top()->word_end) && nodes.top()->children.empty()) {
+            /* code */
+            nodes.pop();
+            nodes.top()->children.erase(word.back());
+            word.pop_back();
+        }
+    }
+
+    /**
+     * @brief helper function to predict/recommend words that starts with a
+     * given prefix from the end of prefix's node iterate through all the child
+     * nodes by recursively appending  all the possible words below the trie
+     * @param prefix string to recommend the words
+     * @param element node at the end of given prefix
+     * @param results list to store the all possible words
+     * @returns list of recommended words
+     */
+    std::vector<std::string> get_all_words(std::vector<std::string> results,
+                                           const std::shared_ptr<Node>& element,
+                                           std::string prefix) {
+        if (element->word_end) {
+            results.push_back(prefix);
+        }
+        if (element->children.empty()) {
+            return results;
+        }
+        for (auto const& x : element->children) {
+            std::string key = "";
+            key = x.first;
+            prefix += key;
+
+            results =
+                get_all_words(results, element->children[x.first], prefix);
+
+            prefix.pop_back();
+        }
+
+        return results;
+    }
+
+    /**
+     * @brief predict/recommend a words that starts with a given prefix
+     * @param prefix string to search for
+     * @returns list of recommended words
+     */
+    std::vector<std::string> predict_words(const std::string& prefix) {
+        std::vector<std::string> result;
+        std::shared_ptr<Node> curr = root_node;
+        /* treversing till the end of given prefix in trie*/
+
+        for (char ch : prefix) {
+            if (curr->children.find(ch) == curr->children.end()) {
+                return result;
+            }
+
+            curr = curr->children[ch];
+        }
+
+        /* if given prefix is the only word without children */
+        if (curr->word_end && curr->children.empty()) {
+            result.push_back(prefix);
+            return result;
+        }
+
+        result = get_all_words(
+            result, curr,
+            prefix);  ///< iteratively and recursively get the recommended words
+
+        return result;
+    }
+};
+
+}  // namespace data_structures
+
+/**
+ * @brief Self-test implementations
+ * @returns void
+ */
+static void test() {
+    data_structures::Trie obj;
```

Comment:
```suggestion
    data_structures::trie_using_hashmap::Trie obj;
```

Reply:
Hi, I had Updated the code, but some checks are failing with this error in workflow
![error1](https://user-images.githubusercontent.com/25405739/111177160-5bf4a580-8580-11eb-8654-50d544e06fb5.PNG)